### PR TITLE
Add test for documented configuration examples

### PR DESCRIPTION
**Pull request by jarhodes314**
## Proposed changes

More #1812 changes. This adds testing for the example values provided using doku and a way to infer the known configuration keys from doku (which forms the groundwork for the tedge config changes that will follow in the hopefully no-too-distant future).

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
#1812

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -18,7 +18,7 @@ resolver = "2"
 version = "0.10.0"
 authors = ["thin-edge.io team <info@thin-edge.io>"]
 edition = "2021"
-rust-version = "1.64"
+rust-version = "1.65"
```
Comment (jarhodes314) [inline Cargo.toml:21]: I had a convenient use case for let-else in the tests, but I can easily change this to match if this version bump is unwanted.

Comment (jarhodes314) [reply Cargo.toml:21]: I've created https://github.com/thin-edge/thin-edge.io/pull/1885 in an effort to fix the integration test workflow as that will need to be merged separately to get the integration tests to run here.

File: crates/common/tedge_config/src/tedge_config_cli/tedge_config_dto.rs [hunk 2]
```diff
@@ -227,3 +225,77 @@ pub struct ServiceTypeConfigDto {
...
+        for (key, ty) in struct_field_paths(None, &fields) {
+            verify_examples_for::<TEdgeConfigDto>(&key, ty)
+        }
+    }
+
+    fn verify_examples_for<Dto>(key: &str, ty: doku::Type)
+    where
+        Dto: Default + Serialize + DeserializeOwned,
+    {
+        for example in ty.example.iter().flat_map(|e| e.iter()) {
```
Comment (didier-wenzek) [inline crates/common/tedge_config/src/tedge_config_cli/tedge_config_dto.rs:253]: Thanks for this exhaustive test of all the examples. I tried it and yes, indeed, this catches mistakes in the doc.

**Review by didier-wenzek — APPROVED** [review]
Approved

**Pull request merged**